src/api/main/graph.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import {type WeightedGraph} from "../../core/weightedGraph";
+import {Ledger} from "../../core/ledger/ledger";
+import type {PluginDirectoryContext} from "../plugin";
+import {type PluginId} from "../pluginId";
+import {Plugin} from "../plugin";
+import {type TaskReporter, SilentTaskReporter} from "../../util/taskReporter";
+import {CascadingReferenceDetector} from "../../core/references/cascadingReferenceDetector";
+import {type ReferenceDetector} from "../../core/references/referenceDetector";
+import {ensureIdentityExists} from "../../core/ledger/identityProposal";
+
+export type GraphInput = {|
+  +plugins: Array<{|
+    +plugin: Plugin,
+    +directoryContext: PluginDirectoryContext,
+  |}>,
+  +ledger: Ledger,
+|};
+
+export type GraphOutput = {|
+  +pluginGraphs: $ReadOnlyArray<{|
+    +pluginId: PluginId,
+    +weightedGraph: WeightedGraph,
+  |}>,
+  +ledger: Ledger,
+|};
+
+/**
+  
+
+  Might mutate the ledger that is passed in.
+ */
+export async function graph(
+  input: GraphInput,
+  scope: $ReadOnlyArray<PluginId>,
+  taskReporter: TaskReporter = new SilentTaskReporter()
```

Comment:
Do you suggest forcing the API user to use a task reporter? or making this param optional? I don't think a task reporter should be required to use this API, especially since we don't expose our own task reporters in the API. A silent task reporter seemed easier than adding a bunch of if statements, or constructing a noop task reporter.